app/ControllerLog.php

```diff
@@ -106,6 +106,38 @@ public static function aggregateAllControllersByPosAndMonth($year = null, $month
         }, []);
     }
 
+    public static function aggregateAllControllersByPosAndQuarter($year = null, $month = null) {
+        $query = static::query()
+            ->rightJoin('roster', function ($join) use ($year, $month) {
+                $join->on('roster.id', '=', 'controller_log.cid');
+                if ($year != null) {
+                    $join->where(DB::Raw("date_format(STR_TO_DATE(`date`, '%m/%d/%Y'), '%y')"), 'like', $year);
+                }
+                if ($month != null) {
+                    $join->where(function ($query) use ($month) {
+                        $quarter = ceil($month / 3);
+                        $first_month_of_qtr = $quarter * 3 - 2;
+                        $query->where(DB::Raw("date_format(STR_TO_DATE(`date`, '%m/%d/%Y'), '%c')"), 'like', $first_month_of_qtr)
```

Comment:
I don't understand what this massive query is doing - I'd appreciate a comment or two explaining the general logic, for when we're back in a year and a half, this isn't working, and we have to decipher it

Reply:
The new query requires that we first define what a quarter is. This code determines what the first month of the selected quarter is.